AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt

```diff
@@ -105,7 +105,7 @@ class SharedDecksDownloadFragment : Fragment() {
         mTryAgainButton = view.findViewById(R.id.try_again_deck_download)
         mCheckNetworkInfoText = view.findViewById(R.id.check_network_info_text)
 
-        val fileToBeDownloaded = arguments?.getSerializable(DOWNLOAD_FILE) as DownloadFile
+        val fileToBeDownloaded = arguments?.getSerializableCompat<Serializable>(DOWNLOAD_FILE) as DownloadFile
```

Comment:
If you read the code, the fragment should always be initialised

* with arguments
* which include `DOWNLOAD_FILE`

It is an error otherwise. I'm happy to fail at this point if the contract isn't satisfied.

```suggestion
        val fileToBeDownloaded = requireArguments().getSerializableCompat<DownloadFile>(DOWNLOAD_FILE)!!
```